src/components/application_manager/src/hmi_capabilities_impl.cc

```diff
@@ -567,27 +593,23 @@ HMICapabilitiesImpl::active_tts_language() const {
 
 void HMICapabilitiesImpl::set_ui_supported_languages(
     const smart_objects::SmartObject& supported_languages) {
-  if (ui_supported_languages_) {
-    delete ui_supported_languages_;
-  }
-  ui_supported_languages_ = new smart_objects::SmartObject(supported_languages);
+  smart_objects::SmartObjectSPtr new_value =
+      std::make_shared<smart_objects::SmartObject>(supported_languages);
+  ui_supported_languages_.swap(new_value);
```

Comment:
@LitvinenkoIra I suggest to consider the next behavior:
 ```
ui_supported_languages_.swap(std::make_shared<smart_objects::SmartObject>(supported_languages));
```

Reply:
@IGapchuk I think this line will be too long then, propose to lease as is